nbformat/validator.py

```diff
@@ -241,21 +242,35 @@ def validate(nbjson, ref=None, version=None, version_minor=None, relax_add_props
     Raises ValidationError if not valid.
     """
     if version is None:
-        from .reader import get_version
-        (version, version_minor) = get_version(nbjson)
+        version, version_minor = get_version(nbjson)
+
+    try:
+        err = next(iter_validate(nbjson, ref=ref, version=version, version_minor=version_minor, relax_add_props=relax_add_props))
```

Comment:
I think we could follow the pattern in jsonschema itself, and do:

```python
# raise the first validation error if there was one
for error in iter_validate(...):
    raise error
```

to raise the first error without the next/try/except logic.